core_impl/src/main/java/com/google/instrumentation/trace/SpanImpl.java

```diff
@@ -303,20 +330,45 @@ public void end(EndSpanOptions options) {
     void onEnd(SpanImpl span);
   }
 
+  private static final class AttributesWithCapacity extends LinkedHashMap<String, AttributeValue> {
+    private int capacity;
+    private int totalRecordedAttributes = 0;
+    private static final long serialVersionUID = 42L;
```

Comment:
This field is a little misleading if the class is never serialized.  I think it would help to have a comment.